component/src/governance/check.rs

```diff
@@ -291,4 +261,81 @@ pub mod stateful {
 
         Ok(())
     }
+
+    pub async fn proposal_withdraw(
+        state: &State,
+        proposal_withdraw: &ProposalWithdraw,
+    ) -> Result<()> {
+        // Any voteable proposal can be withdrawn
+        proposal_voteable(state, proposal_withdraw.proposal).await?;
+        Ok(())
+    }
+
+    pub async fn proposal_deposit_claim(
+        state: &State,
+        proposal_deposit_claim: &ProposalDepositClaim,
+    ) -> Result<()> {
+        // Any finished proposal can have its deposit claimed
+        proposal_claimable(state, proposal_deposit_claim.proposal).await?;
+        // Check that the deposit amount matches the proposal being claimed
+        proposal_claim_valid_deposit(
+            state,
+            proposal_deposit_claim.proposal,
+            proposal_deposit_claim.deposit_amount,
+        )
```

Comment:
These two checks make sure that (1) the proposal is in the "finished" state (not "voting", "withdrawn" (which means "withdrawn but not yet finished voting"), or "claimed") precisely, and (2) the deposit amount matches the amount put in when the proposal was submitted. Very important second check!